src/Workspaces/Core/Portable/FindSymbols/SearchQuery.cs

```diff
@@ -7,7 +7,7 @@
 
 namespace Microsoft.CodeAnalysis.FindSymbols
 {
-    internal class SearchQuery : IDisposable
+    internal readonly struct SearchQuery : IDisposable
```

Comment:
was very close to being able to make the whole host of these into ref-struct types.  But SearchQuery itself is captured into the heap through teh use of a "predicate" delegate it hands out to the compiler to do some work.  So for now these have to be normal structs unfortunately.